### PR TITLE
create deck endpoint/db create script

### LINKED ISSUE #23: Story - create /decks endpoint
Support the following calls and make sure this endpoint can only be reached when authenticated (pass along jwt)

- POST 

request body contains:
```
{
  "username": <username>,
  "deck_name": <deckname>,
  "deck_list: [
    1,
    2,
    3,
    ...
   ]
}
```

-  GET no request body called like - GET <app_url>/deck/{id}

### PR DESCRIPTION
**Pull request by s32989**

### CODE REVIEW

File: be/app/src/main/java/com/deckbop/app/dao/DeckDAO.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+@Component
+public class DeckDAO {
+    
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+    
+    public void createDeck(DeckDTO deckDto){
+        Deck deck = new Deck();
+//        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```
Comment (cobbettn) [inline be/app/src/main/java/com/deckbop/app/dao/DeckDAO.java:24]: we might want to discuss if we want to pull in the user id on the backend or as part of the request

File: be/app/src/main/java/com/deckbop/app/model/Deck.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    String deckName;
+    
+    public Deck(int deckId, int userId, int deckListId, String deckName){
+        this.deckId = deckId;
+        this.userId = userId;
+        this.deckListId = deckListId;
+        this.deckName = deckName;
+    }
+    
+    public Deck(){
```
Comment (cjn98) [inline be/app/src/main/java/com/deckbop/app/model/Deck.java:18]: needed?

File: be/app/src/main/java/com/deckbop/app/model/Deck.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        
+    }
+
+    public int getDeckId() {
```
Comment (cobbettn) [inline be/app/src/main/java/com/deckbop/app/model/Deck.java:18]: remoge?

File: be/app/pom.xml [hunk 4]
```diff
@@ -68,6 +68,10 @@
 			<artifactId>spring-security-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+                <dependency>
```
Comment (cobbettn) [inline be/app/pom.xml:71]: spacing

**Review by cobbettn — APPROVED** [review]

**Pull request merged**